src/components/UserContext/UserContext.tsx

```diff
@@ -0,0 +1,137 @@
+import React, { useState, useCallback } from 'react';
+import { getTodos, getUser } from '../../api';
+import { SortTypes } from '../../types/SortTypes';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { debounce } from '../utils/debounce';
+
+type Context = {
+  todos: Todo[],
+  selectedUser: number,
+  selectedTodoId: number,
+  userInfo: User | null,
+  sortType: SortTypes,
+  query: string,
+  appliedQuery: string,
+  isLoaded: boolean,
+  isError: boolean,
+
+  getUserFromServer: () => void,
+  handleClose: () => void,
+  getTodosFromServer: () => void,
+  changeSortType: (event: React.ChangeEvent<HTMLSelectElement>) => void,
+  applyQuery: ((arg: string) => void),
+  handleSearch: (event: React.ChangeEvent<HTMLInputElement>) => void,
+  clearSearchBar: () => void,
+  selectCurrentTodo: (id: number, userId: number) => void
+};
+
+export const UserContext = React.createContext<Context>({
+  todos: [],
```

Comment:
userContext shouldn't know anything about todo info and function for managing todos
let's create 2 separate context: one for user data and functions for handling each changes, related to user
and second one for the same with todo data

in this way will be easier to add management function for separate models (for user and for todo), and it'll allow use todo or user provider separately of each other

Reply:
hope I understand :)